packages/apollo-collaboration-server/src/fileHandling/fileHandling.controller.ts

```diff
@@ -224,4 +162,47 @@ export class FileHandlingController {
   getFastaInfo(@Headers() headers) {
     return this.fileService.getFastaInfo()
   }
+
+  /**
+   * Save new uploaded file into local filesystem and then loads it into cache. The filename in local filesystem will be: 'uploaded' + timestamp in ddmmyyyy_hh24miss -format + original filename
+   * You can call this endpoint like: curl http://localhost:3000/fileHandling/uploadtocache -F 'file=@./save_this_file.txt' -F 'name=test'
+   * @param file File to save
+   * @returns Return status 'HttpStatus.OK' if save was successful
+   * or in case of error return throw exception
+   */
+  @UseGuards(JwtAuthGuard)
+  @Post('/uploadtocache')
+  @UseInterceptors(FileInterceptor('file'))
+  uploadFile(@UploadedFile() file: Express.Multer.File) {
+    const tmpFileName = this.fileService.saveNewFile(file)
+    this.fileService.loadGFF3FileIntoCache(tmpFileName)
+  }
+
+  /**
+   * Download cache. First write cache into file and then download the file
+   * @param res
+   * @returns
+   */
+  @UseGuards(JwtAuthGuard)
+  @Get('/downloadcache')
+  downloadCache(@Res() res: Response) {
+    this.logger.debug('Starting to write cache into file...')
+    this.fileService.downloadCacheAsGFF3file().then((msg) => {
+      this.logger.debug(`Now downloading file =${msg}`)
+      const file = createReadStream(msg)
+      return file.pipe(res)
```

Comment:
Should use `StreamableFile`, similar to `getFile`

Reply:
Fixed